jail/20230308.md

```diff
@@ -0,0 +1,176 @@
+Useful for WiFi drivers. Also available on OpenBSD. Should also support `/etc/stop_if.<device>`
+
+If the `/etc/start_if.⟨interface⟩` file is present, it is read and executed by the `sh(1)` interpreter before configuring the interface as specified in the `ifconfig_⟨interface⟩` and  
+`ifconfig_⟨interface⟩_alias⟨n⟩` variables.
+
+**Dave**: Two broad directions/initiatives: Build a list of desired in-kernel Jail resources/functionality. Seek development and funding. Both need consensus. Specifically, the state machine mentioned above for cleanly correcting interrupted states. When ZFS support arrived in Jails, there was metadata for dataset etc. The kernel interface supports additional metadata. The kernel could issue a UUID that will not be reused and cannot conflict. Currently facing the risk of race conditions. Would like userspace Jails, created by unprivileged users. "Rootless Jails" Linux does this and it has various security implications. Igor has notes and thoughts on this. Some amount of restriction would be required. Could have certain sysctls disabled, restrict hard links... Could use the MAC framework.
+
+**Antranig** : Re: Funding: His company could assist but is an Elixir, rather than C shop. Rootless jails: We have rootless chroots, no? Yes. This may be a starting place. He's using a variable UUID in the `jail.conf` to handle this. Note how Zone restart behavior differs.
+
+How much promise does [runj](https://github.com/samuelkarp/runj) have?
+
+dfr's implementation is much further ahead (podman/buildah etc) than runj.
+
+Note the Open Container Initiative and [OCI for FreeBSD](https://medium.com/@dfr/oci-containers-for-freebsd-512a6df2bc85)
+
+Notes on dfr'spodman port
+
+Goran's [current patch](https://github.com/mekanix/freebsd-src/tree/feature/jail) Patches the jail utility which is aware of all applicable configuration paths.  
+Note "wildcard" jails - global config and other jails can override one or more above it.  
+`/etc/jail.conf` could be at the end... Start with `jail.conf.d`? Thoughts on the order of the  
+parser?
+
+**Dan**: Sounds like /etc/defaults jail.conf is really jail.conf.defaults that you manually  
+override. "This is a setting you cannot override" might be desired.  
+**Note**: ` * { exec.start = ""; exec.stop = ""; }`
+
+**Note:** `/etc/jail.<name>.conf`  
+**Note:** `jail -r -f /path/to/jail.conf '*'`
+
+**Dave**: The kernel, not configuration files should manage the true state of Jails. Need a description syntax that is handed to the kernel. Note [UCL/libucl](https://github.com/vstakhov/libucl). Not sure if libucl supports variables but perhaps it can be added. Is there any UCL support in Jail yet?
+
+Note that `jls` supports `--libxo=json` etc.
+
+If you have an integer for a Jail name, it will use that as the jail ID. Feature or bug? What happens when they collide?
```

Comment:
This is very intentional, and there's no such thing as a collision -- the jid can't be auto-assigned to a jail if it's been taken by a jid-named-jail.